middlewares/metrics.go

```diff
@@ -0,0 +1,39 @@
+package middlewares
+
+import (
+	"net/http"
+	"time"
+	"strconv"
+	"github.com/go-kit/kit/metrics"
+	)
+
+type Metrics interface {
+	getReqsCounter() metrics.Counter
+	getLatencyHistogram() metrics.Histogram
+	handler() http.Handler
+}
+
+type MetricsWrapper struct {
+	Impl Metrics
+}
+
+func NewMetricsWrapper(impl Metrics) *MetricsWrapper {
+	var metricsWrapper = MetricsWrapper {
+		Impl:	impl,
+	}
+
+	return &metricsWrapper
+}
+
+func (m *MetricsWrapper) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	start := time.Now()
+	prw := &responseRecorder{rw, http.StatusOK}
+	next(prw , r)
+	labels := []string{ "code", strconv.Itoa(prw.StatusCode()), "method", r.Method, "path", r.URL.Path }
```

Comment:
I'm not sure the request path is suitable as a label for a generic metrics handler, because it has an unbounded cardinality.